### PR TITLE
announce 4.28

### PR DESCRIPTION
**Pull request by jjergus**

### CODE REVIEW

File: _posts/2019-10-23-hhvm-4.28.markdown [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  with namespaces.
+- A new function,
+  [`set_pre_timeout_handler`](https://docs.hhvm.com/hack/reference/function/set_pre_timeout_handler/),
+  was added. The pre-timeout handler allows cheaper debugging of timeouts by
+  invoking analysis code a few seconds prior to requests timing out. It also
+  allows re-registering a new pre-timeout callback much like the memory
+  threshold callbacks.
+- Fixed a bug that would cause `hackfmt` to reformat code in a way that breaks
+  `HH_FIXME` comments in some cases.
+- It is no longer a type error to reference the built-in type
```
Comment (fredemmott) [inline _posts/2019-10-23-hhvm-4.28.markdown]: <3

File: _posts/2019-10-23-hhvm-4.28.markdown [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  `Awaitable`) will no longer override the built-in type. Such types can now
+  only be referenced using their namespaced name (e.g. `MyNamespace\Vector`, or
+  `namespace\Vector` if in the same namespace).
+- It is now a type error to `use` multiple traits that declare `__construct()`
+  from the same class.
+- Trying to call a non-static method &ldquo;statically&rdquo; will now always
+  trigger a fatal error. This was already true in most cases, except some
+  special cases where it was previously only a warning (e.g. using a non-static
+  method in `array_map(['ClassName', 'nonStaticMethodName'], $arr)`).
+- It is now a type error to pass a value of type `array` where a tuple is
```
Comment (fredemmott) [inline _posts/2019-10-23-hhvm-4.28.markdown]: might need to add a type-error-driven hhast migration for this :( unique code?

Comment (jjergus) [reply _posts/2019-10-23-hhvm-4.28.markdown]: Actually this is just for `array` with no generic types, I should probably make that clear. And `array` with no generic types has been an error (in strict mode) for a long time, so hopefully this isn't too common.

Comment (jjergus) [reply _posts/2019-10-23-hhvm-4.28.markdown]: Basically there's no code that would be valid in strict mode with no FIXMEs that we're breaking.

**Pull request merged**